hicommonwealth/commonwealth
# 7810 spl token gating

## Link to Issue
Closes: #7833

## Description of Changes
- Adds solana balance fetching and batching to TokenBalanceCache
- Enables creation of solana communities(UI/platform)
- Adds UI updates to create Solana SPL requirements on Solana communities 
- Updates schemas and models to include spl balance requirement types 

## Test Plan
- run migration and yarn
- sign-in with Phantom Wallet
- Make yourself admin of the solana community via DB
- create requirements with the following sample tokens(note sol isnt prefiex with 0x's) using Solana Mainnet Beta
    - Dog wif hat: EKpQGSJtjMFqKZ9KQanSqYXRcF8fBopzLHYxdM65zcjm
    - Jupiter: JUPyiwrYJFskUPiHa7hkeR8VUtAeFoSYbKedZNsDvCN
    - USDC: EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v
- got to the members tab and note that some addresses are included in each group filter
- Set a high value for WIF(ie 1000000) and note that some should not be in group any more

- Additionally test other chain groups(ie evm) to ensure these changes don't create issues/regressions

## Deployment Plan

1. Run migration
2. Manually update private url to solana chainnode

## Other Considerations

- There are not as efficient of batching methods available for solana rpcs atm. I using promise.all on single balance requests is the only current option. This should be Ok for our scale at the current moment but we should monitor for updates to Alchemy rpc to allow for better batching. The current implementation batches with rate-limit in mint but may not be as generally performant as Evm batching
